lib/matplotlib/colors.py

```diff
@@ -1099,14 +1099,37 @@ def from_list(name, colors, N=256, gamma=1.0):
         if not np.iterable(colors):
             raise ValueError('colors must be iterable')
 
-        if (isinstance(colors[0], Sized) and len(colors[0]) == 2
-                and not isinstance(colors[0], str)):
-            # List of value, color pairs
-            vals, colors = zip(*colors)
+        try:
+            # Assume the passed colors are a list of colors
+            # and not a (value, color) tuple.
+            r, g, b, a = to_rgba_array(colors).T
+        except (ValueError, TypeError) as e:
```

Comment:
It pays to be forgiving here and accept `Exception`?  If anything goes wrong in `to_rgba_array` we should do a best-effort check at the other interpretation.

Reply:
I was thinking about this – I think you are right.

Testing different inputs **that are valid `(value, color)` pairs, the only two errors I encountered were `ValueError` and `TypeError`. My reasoning was that if there is a different error raised, checking for `(value, color)` pairs can be skipped.

However, it is very likely that in some edge case, a different Error is raised while still being a correct `(value, color)` tuple. Also, a `ValueError` is probably the most common error raised – and is completely suppressed in the following branch.

I will include this in my next revision.